multidataset_plugin.h

```diff
@@ -0,0 +1,47 @@
+#ifndef MULTIDATASET_PLUGIN_H
+#define MULTIDATASET_PLUGIN_H
+
+//#include <mpi.h>
+#include <stdlib.h>
+#include <string.h>
+#include <hdf5.h>
+#include "H5Timing.h"
+#define ENABLE_MULTIDATASET 0
+#define MULTIDATASET_DEFINE 1
+
+static int max_batch_size_g = 2;
+static int hdf5_method_g = -1;
+static int total_n_events_g = -1;
+
+
+#if MULTIDATASET_DEFINE == 1
+typedef struct H5D_rw_multi_t
+{
+    hid_t dset_id;          /* dataset ID */
+    hid_t dset_space_id;    /* dataset selection dataspace ID */
+    hid_t mem_type_id;      /* memory datatype ID */
+    hid_t mem_space_id;     /* memory selection dataspace ID */
+    union {
+        void *rbuf;         /* pointer to read buffer */
+        const void *wbuf;   /* pointer to write buffer */
+    } u;
+} H5D_rw_multi_t;
+#endif
+
+typedef struct multidataset_array {
+    std::vector<hsize_t> *start;
+    std::vector<hsize_t> *end;
+    hsize_t last_end;
+    hid_t did;
+    hid_t mtype;      /* memory datatype ID */
+    std::vector<char*> *temp_mem;
+} multidataset_array;
```

Comment:
It would be better of all of these vectors were just held by value instead of by pointer. The compiler would then handle all the memory management.